### PR TITLE
add freshwater research station and associated terrain/furniture.

### PR DESCRIPTION
**Pull request by curstwist**
#### Summary

SUMMARY: Content "Adds a new lake special, submerged freshwater research station and supporting terrains/furniture."

#### Purpose of change

Begin taking advantage of the lake z levels and hopefully encourage others to work on the accompanying mechanics.

#### Describe the solution

Adds a small civilian research facility in the lake.   This design offers a surface access area though future ones could be fully submerged.  We don't have the mechanics yet for navigating water between z levels and this allows it to be playable.  It is also a fairly common design for underwater labs.

Added terrains and furniture:
t_metal_pontoon_dp
t_metal_dock_deep
t_metal_dock_deep_frame
t_concrete_dock_deep_pile
t_metal_dock

t_wall_bulkhead
t_door_metal_bulkhead_o
t_door_metal_bulkhead_c
t_porthole
t_bulkhead_floor
t_bulkhead_roof

t_lake_bed_concrete_yellow
t_lake_bed_concrete

f_cleat (small metal boat tie item)
f_foot_locker
f_chest
f_control_station (generic per Erk's request)

I've also added a moon pool entrance and a airlock hatch entrance for further mechanics testing/development.  

#### Describe alternatives you've considered
I considered adding more to the facility using nested mapgen, but that can come at a later stage.  This PR already has plenty of content to review.

#### Testing
Spawned in world.  Made sure it can spawn naturally.  I'm open to making this a debug spawn only special until water if further developed but I also made it workable with the current limitations.

I ran into a load error with scuba tanks that I'm including here.  Scuba tanks are so rare, I suspect this already existed.  This location offers plenty of tanks to test with to resolve the error.  If it's something I can fix, let me know : ).

Note, I've tested this with charges_min & max and that allowed the tanks to spawn with charges but a save/reload resulted in the error message triggering.

scuba tank error

#### Additional context
This version is a civilian facility so it doesn't offer anything too crazy but it'll be a reliable location for water content like tanks and boats.

surface:
freshwater_research_station_surface

-1 
freshwater_research_station_1 and 2 below

-2
freshwater_research_station_2 below

-3
freshwater_research_station_3 below

-4
freshwater_research_station_4 below

-5 (I've added underwater cement varieties for the foundation pad.  yellow painted cement outlines the moon pool entrance)
freshwater_research_station_lakebed

### CODE REVIEW

**Comment by aoae23** [comment]
I don't think the load error is related to scuba tanks. I've seen it pop up randomly with other items that have charges.

**Comment by John-Candlebury** [comment]
This is one of the coolest maps I've seen in a long time.

File: data/json/furniture_and_terrain/terrain-bridges-docks.json [hunk 1]
```diff
@@ -216,5 +216,108 @@
...
+    "type": "terrain",
+    "id": "t_metal_dock",
+    "name": "dock",
+    "description": "A metal platform held by a support made of concrete set into the ground.",
+    "symbol": "8",
+    "looks_like": "t_metal_floor",
+    "color": "gray",
+    "move_cost": 2,
+    "flags": [ "TRANSPARENT", "FLAT" ],
+    "deconstruct": { "ter_set": "t_water_sh", "items": [ { "item": "nail", "charges": [ 6, 12 ] }, { "item": "2x4", "count": 8 } ] },
```
Comment (Maleclypse) [inline data/json/furniture_and_terrain/terrain-bridges-docks.json:230]: The deconstruct and smash items don't match up?

File: data/json/furniture_and_terrain/terrain-bridges-docks.json [hunk 2]
```diff
@@ -216,5 +216,108 @@
...
+    "type": "terrain",
+    "id": "t_concrete_dock_deep_pile",
+    "name": "deep dock pile",
+    "description": "A sturdy pile made of poured concrete extending all the way to the riverbed.  A metal frame will be required to support a surface.",
+    "looks_like": "t_concrete_wall",
+    "symbol": "|",
+    "color": "gray",
+    "move_cost": 8,
+    "flags": [ "TRANSPARENT", "LIQUID", "SWIMMABLE", "DEEP_WATER", "FISHABLE" ],
+    "deconstruct": { "ter_set": "t_water_moving_dp", "items": [ { "item": "spike", "count": 4 }, { "item": "rock", "count": [ 6, 10 ] } ] },
```
Comment (Maleclypse) [inline data/json/furniture_and_terrain/terrain-bridges-docks.json:250]: Would rebar replace spikes or be in addition to spikes? Rebar might also not be used in concrete piles.

File: data/json/furniture_and_terrain/terrain-bridges-docks.json [hunk 3]
```diff
@@ -216,5 +216,108 @@
...
+    "id": "t_metal_dock_deep",
+    "name": "deep dock",
+    "description": "A metal platform held by a tall metal pile that extends to the riverbed.  Very sturdy, and likely to outlast you.",
+    "symbol": "8",
+    "color": "gray",
+    "move_cost": 2,
+    "flags": [ "TRANSPARENT", "FLAT" ],
+    "deconstruct": {
+      "ter_set": "t_metal_dock_deep_frame",
+      "items": [ { "item": "nail", "charges": [ 12, 24 ] }, { "item": "sheet_metal", "count": 4 } ]
```
Comment (Maleclypse) [inline data/json/furniture_and_terrain/terrain-bridges-docks.json:291]: Are the nails standing in for metal screws?

Comment (curstwist) [reply data/json/furniture_and_terrain/terrain-bridges-docks.json:291]: yes, we don't have screws (unless added recently)

File: data/json/furniture_and_terrain/terrain-bridges-docks.json (hunk 2 above)
Comment (curstwist) [reply data/json/furniture_and_terrain/terrain-bridges-docks.json]: updated

File: data/json/furniture_and_terrain/terrain-bridges-docks.json [hunk 4]
```diff
@@ -216,5 +216,118 @@
...
     }
+  },
+  {
+    "type": "terrain",
+    "id": "t_metal_dock",
+    "name": "dock",
+    "description": "A metal platform held by a support made of concrete set into the ground.",
+    "symbol": "8",
+    "looks_like": "t_metal_floor",
+    "color": "gray",
```
Comment (ZhilkinSerg) [inline data/json/furniture_and_terrain/terrain-bridges-docks.json:227]: Should be either `light_gray` or `dark_gray`.

File: data/json/furniture_and_terrain/terrain-bridges-docks.json [hunk 5]
```diff
@@ -216,5 +216,118 @@
...
+    }
+  },
+  {
+    "type": "terrain",
+    "id": "t_concrete_dock_deep_pile",
+    "name": "deep dock pile",
+    "description": "A sturdy pile made of poured concrete extending all the way to the riverbed.  A metal frame will be required to support a surface.",
+    "looks_like": "t_concrete_wall",
+    "symbol": "|",
+    "color": "gray",
```
Comment (ZhilkinSerg) [inline data/json/furniture_and_terrain/terrain-bridges-docks.json:250]: Should be either `light_gray` or `dark_gray`.

File: data/json/furniture_and_terrain/terrain-bridges-docks.json [hunk 6]
```diff
@@ -216,5 +216,118 @@
...
+      ]
+    }
+  },
+  {
+    "type": "terrain",
+    "id": "t_metal_dock_deep_frame",
+    "name": "deep dock frame",
+    "description": "A metal frame secured atop a concrete pile.  Adding a metal surface will make this into a proper dock section.",
+    "symbol": "#",
+    "color": "gray",
```
Comment (ZhilkinSerg) [inline data/json/furniture_and_terrain/terrain-bridges-docks.json:276]: Should be either `light_gray` or `dark_gray`.

File: data/json/furniture_and_terrain/terrain-bridges-docks.json [hunk 7]
```diff
@@ -216,5 +216,118 @@
...
+      "items": [ { "item": "pipe", "count": 4, "prob": 25 }, { "item": "scrap", "count": [ 4, 8 ] } ]
+    }
+  },
+  {
+    "type": "terrain",
+    "id": "t_metal_dock_deep",
+    "name": "deep dock",
+    "description": "A metal platform held by a tall metal pile that extends to the riverbed.  Very sturdy, and likely to outlast you.",
+    "symbol": "8",
+    "color": "gray",
```
Comment (ZhilkinSerg) [inline data/json/furniture_and_terrain/terrain-bridges-docks.json:296]: Should be either `light_gray` or `dark_gray`.

File: data/json/furniture_and_terrain/terrain-bridges-docks.json [hunk 8]
```diff
@@ -216,5 +216,118 @@
...
+    }
+  },
+  {
+    "type": "terrain",
+    "id": "t_metal_pontoon_dp",
+    "name": "metal floating dock",
+    "description": "A floating temporary dock, anchored to keep it from drifting away.",
+    "looks_like": "t_grate",
+    "symbol": "8",
+    "color": "gray",
```
Comment (ZhilkinSerg) [inline data/json/furniture_and_terrain/terrain-bridges-docks.json:320]: Should be either `light_gray` or `dark_gray`.

File: data/json/furniture_and_terrain/terrain-doors.json [hunk 9]
```diff
@@ -2420,5 +2420,49 @@
...
     "copy-from": "t_door_glass_lab_o"
+  },
+  {
+    "type": "terrain",
+    "id": "t_door_metal_bulkhead_c",
+    "name": "closed bulkhead door",
+    "description": "An extremely resilient door of cast metal, it is watertight and opened using the fly wheel.",
+    "symbol": "+",
+    "looks_like": "t_door_metal_c",
+    "color": "gray",
```
Comment (ZhilkinSerg) [inline data/json/furniture_and_terrain/terrain-doors.json:2431]: Should be either `light_gray` or `dark_gray`.

File: data/json/furniture_and_terrain/terrain-doors.json [hunk 10]
```diff
@@ -2420,5 +2420,49 @@
...
+    }
+  },
+  {
+    "type": "terrain",
+    "id": "t_door_metal_bulkhead_o",
+    "name": "open bulkhead door",
+    "description": "An extremely resilient door of cast metal, it is watertight and opened using the fly wheel.",
+    "symbol": "+",
+    "looks_like": "t_door_metal_o",
+    "color": "gray",
```
Comment (ZhilkinSerg) [inline data/json/furniture_and_terrain/terrain-doors.json:2453]: Should be either `light_gray` or `dark_gray`.

File: data/json/furniture_and_terrain/terrain-liquids.json [hunk 11]
```diff
@@ -311,5 +311,85 @@
...
+  },
+  {
+    "type": "terrain",
+    "id": "t_bulkhead_roof",
+    "//": "for eventual use with water z levels.  Currently non-functional.  Roof for submerged structures.",
+    "name": "bulkhead",
+    "description": "The top of a submerged structure.  With a watertight container, you could gather fresh water from here.  Not safe to drink as is.",
+    "symbol": "#",
+    "looks_like": "t_metal_flat_roof",
+    "color": "gray",
```
Comment (ZhilkinSerg) [inline data/json/furniture_and_terrain/terrain-liquids.json:323]: Should be either `light_gray` or `dark_gray`.

File: data/json/furniture_and_terrain/terrain-liquids.json [hunk 12]
```diff
@@ -311,5 +311,85 @@
...
+  },
+  {
+    "type": "terrain",
+    "id": "t_bulkhead_floor",
+    "//": "for eventual use with water z levels.  Currently non-functional.  Roof for submerged structures.",
+    "name": "bulkhead",
+    "description": "The bottom of a submerged structure.",
+    "symbol": "#",
+    "looks_like": "t_metal_floor",
+    "color": "gray",
```
Comment (ZhilkinSerg) [inline data/json/furniture_and_terrain/terrain-liquids.json:337]: Should be either `light_gray` or `dark_gray`.

**Comment by cinead** [comment]
Thank you for writing this! 

Re: "Begin taking advantage of the lake z levels and hopefully encourage others"

I am working on some json for aquatic and submerged plant life that should compliment this well.

Thanks for the encouragement.

**Pull request merged**